client/service/src/client/client.rs

```diff
@@ -896,39 +896,29 @@ where
 		operation.op.mark_finalized(BlockId::Hash(block), justification)?;
 
 		if notify {
-			let finalized =
-				route_from_finalized.enacted().iter().map(|elem| elem.hash).collect::<Vec<_>>();
+			let finalized = std::iter::once(last_finalized)
```

Comment:
Is correct to include the `last_finalized`? I didn't included it to prevent a double notification for that block:
- 1st notification when it is the last finalized block (explicit finalization)
- 2nd notification when it s the fist implicitly finalized block

Indeed here now a2 is repeated twice:
https://github.com/paritytech/substrate/blob/9ce46c344419d7dee25f01c24219d078ca459100/client/service/test/src/client/mod.rs#L1326-L1328

Reply:
In this way if blocks are (as it is usually the "normal" case) finalized one after the other then for each finalized block you'll always receive two finalizations (currently finalized one + prev finalized one)